### PR TITLE
PLUGIN-170-Datastore-auth-update

### PR DESCRIPTION
**Pull request by ailegion**
Added: Ability to provide Service Account type as JSON for Datastore Plugins

JIRA: Jira Ticket: https://issues.cask.co/browse/PLUGIN-170

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/datastore/sink/util/DatastoreSinkConstants.java [hunk 1]
```diff
@@ -34,7 +34,8 @@
   String CONFIG_PROJECT = "mapred.gcd.output.project";
   String CONFIG_USE_AUTOGENERATED_KEY = "mapred.gcd.output.use.autogenerated.key";
   String CONFIG_BATCH_SIZE = "mapred.gcd.output.batch.size";
-  String CONFIG_SERVICE_ACCOUNT_FILE_PATH = "mapred.gcd.output.service.account.file.path";
+  String CONFIG_SERVICE_ACCOUNT = "mapred.gcd.output.service.account";
+  String CONFIG_SERVICE_ACCOUNT_IS_FILE = "mapred.gcd.output.service.account.isfile";
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/datastore/sink/util/DatastoreSinkConstants.java]: nit: let's change this to CONFIG_SERVICE_ACCOUNT_IS_FILE_PATH

Comment (ailegion) [reply src/main/java/io/cdap/plugin/gcp/datastore/sink/util/DatastoreSinkConstants.java]: changed

File: src/main/java/io/cdap/plugin/gcp/datastore/sink/util/DatastoreSinkConstants.java [hunk 2]
```diff
@@ -34,7 +34,8 @@
   String CONFIG_PROJECT = "mapred.gcd.output.project";
   String CONFIG_USE_AUTOGENERATED_KEY = "mapred.gcd.output.use.autogenerated.key";
   String CONFIG_BATCH_SIZE = "mapred.gcd.output.batch.size";
-  String CONFIG_SERVICE_ACCOUNT_FILE_PATH = "mapred.gcd.output.service.account.file.path";
+  String CONFIG_SERVICE_ACCOUNT = "mapred.gcd.output.service.account";
+  String CONFIG_SERVICE_ACCOUNT_IS_FILE_PATH = "mapred.gcd.output.service.account.isfile";
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/datastore/sink/util/DatastoreSinkConstants.java]: the string also needs to change to indicate that it's a file path. something like " mapred.gcd.output.service.account.isfilepath"

**Review by rmstar — APPROVED** [review]
Please squash commits after addressing the minor comment.

**Comment by ailegion** [comment]
> Please squash commits after addressing the minor comment.

fixed and squashed

**Review by rmstar — APPROVED** [review]

**Pull request merged**